ArduPlane/altitude.cpp

```diff
@@ -321,6 +311,46 @@ void Plane::set_target_altitude_proportion(const Location &loc, float proportion
     }
 }
 
+#if AP_TERRAIN_AVAILABLE
+/*
+  change target altitude along a path between two locations
+  (prev_WP_loc and next_WP_loc) where the second location is a terrain
```

Comment:
What about the reverse? What if the prev_WP_loc is terrain and next_WP_loc is not?

Reply:
we handle that with the existing code. It won't follow terrain to get to the destination